src/schema/posts.ts

```diff
@@ -1673,21 +1673,35 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       const mentions = await getMentions(con, content, userId, sourceId);
 
       const pendingPost: CreateSourcePostModeration = {
-        title,
-        content,
         sourceId,
         type,
         sharedPostId,
         externalLink,
         createdById: userId,
       };
 
-      if (commentary) {
+      const isExternal = !!externalLink;
+
+      if (commentary && type === PostType.Share) {
         await validateCommentary(commentary);
-        pendingPost.titleHtml = generateTitleHtml(commentary, mentions);
+        const commentaryHtml = generateTitleHtml(commentary, mentions);
+
+        if (isExternal) {
+          pendingPost.title = title;
+          pendingPost.content = commentary;
+          pendingPost.contentHtml = commentaryHtml;
+        } else {
+          pendingPost.title = commentary;
+          pendingPost.titleHtml = commentaryHtml;
+        }
       }
-      if (content) {
-        pendingPost.contentHtml = markdown.render(content, { mentions });
+
+      if (content && type === PostType.Freeform) {
+        pendingPost.title = title;
+        pendingPost.content = content;
+        pendingPost.contentHtml = markdown
+          .render(content, { mentions })
+          ?.trim();
```

Comment:
For some reason, there is an extra new line.

![Screenshot 2024-11-12 at 12 47 02 PM](https://github.com/user-attachments/assets/542e6609-84de-40db-9b65-605a0dd0bb48)

Reply:
Trim removes whitespaces

Reply:
Yes.